packages/framework-core/src/decorators/read-model.ts

```diff
@@ -18,6 +18,7 @@ export function ReadModel(attributes: RoleAccess): (readModelClass: Class<ReadMo
         class: readModelClass,
         properties: getPropertiesMetadata(readModelClass),
         authorizedRoles: attributes.authorize,
+        before: attributes.before,
```

Comment:
What about doing `before: attributes.before ?? []` ?
This way, we can define the `before` field in the metadata as non-null, simplifying some code (as the one I wrote in THIS comment where the before hooks parameter would not be optional anymore)